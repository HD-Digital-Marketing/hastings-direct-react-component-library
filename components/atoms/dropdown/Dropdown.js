import React from 'react'
import classNames from 'classnames'
import PropTypes from 'prop-types'
import Label from '../label/Label'
import Icon from '../icon/Icon'
import './dropdown.scss'
import chevDown from '../../../src/img/icons/chevron_down.svg'

const Dropdown = props => {

    const {
        options = [],
        name = '',
        label = '',
        theme = '',
        className = ''
    } = props

    const dropdownClass = classNames('a-dropdown', className, {
        'a-dropdown--dark': theme === 'dark'
    })

    return (
        <div className={dropdownClass}>
            {label !== '' ? <Label forLabel={name} theme={theme}>{label}</Label> : ''}
            <div className="a-dropdown__wrap">
                <Icon svgImage={chevDown} svgWidth={15} svgHeight={15} className="a-dropdown__icon"/>
                <select name={name} className="a-dropdown__select">
                    {options.map((option, index) => (
                        <option value={option} key={index}>{option}</option>
                    ))}
                </select>
            </div>
        </div>
    )
}

Dropdown.propTypes = {
    options: PropTypes.array,
    name: PropTypes.string,
    label: PropTypes.string,
    theme: PropTypes.string,
    className: PropTypes.string
}

export default Dropdown
