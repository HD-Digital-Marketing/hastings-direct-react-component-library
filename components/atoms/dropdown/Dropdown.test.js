import React from 'react'
import renderer from 'react-test-renderer'
import Dropdown from './Dropdown'

describe('Check the dropdown renders', () => {
    test('it should match the snapshot', () => {

        const options = [
            'Car policy',
            'Van policy',
            'Bike policy',
            'Home policy',
            'Multicar policy',
            'YouDrive policy'
        ]

        const tree = renderer
            .create(
                <Dropdown theme="dark" options={options} name="choosePolicy" label="Please choose your policy" />
            )
            .toJSON()
        expect(tree).toMatchSnapshot()

    })
})