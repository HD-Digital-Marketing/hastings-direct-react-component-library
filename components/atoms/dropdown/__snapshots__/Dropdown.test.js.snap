// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check the dropdown renders it should match the snapshot 1`] = `
<div
  className="a-dropdown a-dropdown--dark"
>
  <label
    className="a-label a-label--on-dark"
    htmlFor="choosePolicy"
  >
    Please choose your policy
  </label>
  <div
    className="a-dropdown__wrap"
  >
    <span
      className="a-icon a-dropdown__icon"
      style={
        Object {
          "backgroundColor": undefined,
          "borderColor": undefined,
          "borderWidth": 1.5,
          "height": 15,
          "width": 15,
        }
      }
    />
    <select
      className="a-dropdown__select"
      name="choosePolicy"
    >
      <option
        value="Car policy"
      >
        Car policy
      </option>
      <option
        value="Van policy"
      >
        Van policy
      </option>
      <option
        value="Bike policy"
      >
        Bike policy
      </option>
      <option
        value="Home policy"
      >
        Home policy
      </option>
      <option
        value="Multicar policy"
      >
        Multicar policy
      </option>
      <option
        value="YouDrive policy"
      >
        YouDrive policy
      </option>
    </select>
  </div>
</div>
`;
